week3/homework/src/index.js

```diff
@@ -1,3 +1,197 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const express = require('express');
+const fs = require('fs');
+const bodyParser = require('body-parser');
+
+const app = express();
+
+app.use(express.json());
+
+app.use(bodyParser.json());
+app.use(express.urlencoded({ extended: false }));
+
+// Read All Todos
+app.get('/todos', (req, res) => {
+  const db = require('./db');
+  res.status(200).send({
+    msg: 'All todos retrieved successfully',
+    todos: db,
+  });
+});
+
+// Get A Single Todo
+app.get('/todo/:id', (req, res) => {
+  const db = require('./db');
+  const id = parseInt(req.params.id, 10);
+  db.map(todo => {
```

Comment:
map, filter, find are meant to be used with pure functions i.e. functions that do not have side effects
a much nicer way to write this would be

```
const todo = db.find(todo => todo.id === id);
res.status(200).send(todo);
```